test/Concurrency/Runtime/async_task_group_dreamland.swift

```diff
@@ -0,0 +1,87 @@
+//// RUN: %target-run-simple-swift(-Xfrontend -enable-experimental-concurrency) | %FileCheck %s --dump-input always
+//// REQUIRES: executable_test
+//// REQUIRES: concurrency
+//// REQUIRES: OS=macosx
+//// REQUIRES: CPU=x86_64
+//
+//import Dispatch
+//
+//exit(0)
+//
+/////// Reliably and quickly perform work.
+////func work(_ n: Int) async -> Int { n }
+////
+/////// Pretend that the work takes the specified amount of time.
+////func slowWork(_ n: Int, seconds: Int) async -> Int { n }
+////
+/////// Pretend that sometimes the work takes a long time.
+////func randomlySlowWork(_ n: Int) async -> Int { n }
+////
+/////// Throw error.
+////func boom() async -> Int { throw Boom() }
+////
+//
+//// ==== ----------------------------------------------------------------------------------------------------------------
+//// MARK: Select (exactly 1, cancel the rest)
+//// Effectively like Go's select {}
+//
+//// step 1) with existing APIs
+//let first: Int = await Task.Group<Int>.select { group in
+//    await group.add { await work() }
+//    await group.add { await sleep(.seconds(3)) }
+//}
+//
+//// step 2) combining task groups with function builders
+//
+//let first: Int = await Task.Group<Int>.select { group in
+//    async let _ = await work()
+//    async let _ = await sleep(.seconds(3))
+//}
```

Comment:
This would work if we had result builder's `buildDefinition`.
Alternatively, some `async work()` would be nice here to be honest:

```swift
let first: Int = await Task.Group<Int>.select { group in
    async work()
    async sleep(.seconds(3))
}
```

but that's pretty weird perhaps. So anyway, just for future thought, that we could build fairly nice scatter gather APIs with this and just a little more sugar in the future.